test/externalTests/common.sh

```diff
@@ -77,12 +77,24 @@ function setup_solc
 function download_project
 {
     local repo="$1"
-    local solcjs_branch="$2"
-    local test_dir="$3"
-
-    printLog "Cloning $solcjs_branch of $repo..."
-    git clone --depth 1 "$repo" -b "$solcjs_branch" "$test_dir/ext"
-    cd ext
+    local ref_type="$2"
+    local solcjs_ref="$3"
+    local test_dir="$4"
+
+    [[ $ref_type == commit || $ref_type == branch || $ref_type == tag ]] || assertFail
+
+    printLog "Cloning ${ref_type} ${solcjs_ref} of ${repo}..."
+    if [[ $ref_type == commit ]]; then
+        mkdir ext
+        cd ext
+        git init
+        git remote add origin "$repo"
+        git fetch --depth 1 origin "$solcjs_ref"
+        git reset --hard FETCH_HEAD
```

Comment:
Is that really the easiest way to fetch a commit? Can't you just also use ``-b`` with commit hashes or sth like that?

Reply:
But also fine otherwise.

Reply:
Apparently, this is really the way to do this... I now remember looking into this myself at some point being rather stifled by there not being an easy way to do this...

Reply:
It's fine, it's just weird that there's no way to do this more concisely :-).